client/finality-grandpa/src/communication/gossip.rs

```diff
@@ -1121,8 +1136,10 @@ impl<Block: BlockT> Inner<Block> {
 	/// underlying gossip layer, which should happen every 30 seconds.
 	fn round_message_allowed<N>(&self, who: &PeerId, peer: &PeerInfo<N>) -> bool {
 		let round_duration = self.config.gossip_duration * ROUND_DURATION;
-		let round_elapsed = self.round_start.elapsed();
-
+		let round_elapsed = match self.local_view {
+			Some(ref local_view) => local_view.round_start.elapsed(),
+			None => return false,
+		};
```

Comment:
The `None` case together with `self.config.is_authority`, would that give the same behaviour as the old code?

Reply:
Overall there is the following change of behavior here: previously the round start was not behind any kind of optional value, therefore it had to be initialized on gossip creation with `Instant::now()` even though the round had not started yet. With the current approach we will only have round start defined after we have initialized the voter, therefore in the meantime all gossip messages will be disallowed. In practice this shouldn't change anything though, as the voter gets immediately initialized with a round after node startup.